Add new Event type stateless

stveit opened the pull request:
Fix for #346 

Adds new event type called `STATELESS`. This is used instead of `INCIDENT_START` for the initial event when a stateless incident is created. Stateful incident will still start with a `ÌNCIDENT_START` event.

Adds a small test that checks if a stateless event is created when a new stateless incidents is created.

Adds migration files for adding `STATELESS`type and update old entries such that `INCIDENT_START` events that belong to stateless incidents get switched to `STATELESS`.

hmpf commented on tests/incident/test_event.py:
```diff
@@ -195,3 +196,23 @@ def test_event_has_description(self):
+            description=f"Incident #{source_incident_id} created for testing",
+        )
+        event_stateless = incident.events.filter(type=Event.Type.STATELESS)
+        self.assertEqual(1, len(event_stateless))
```
`event_stateless` is a queryset, so do `event_stateless.count()` instead of `len()`. This makes postgres do the count, not python.

hmpf reviewed (changes requested):
Improve the migrations and see other comments.

Also, first letter upper case in commit-messages please. Hm, sounds like a candidate for a pre-commit hook!

hmpf commented:
Oh, and you're working from an outdated master-branch, so there might be merge-shenaningans.

stveit commented:
Added reverse function for migration.
Put all migration in one file.
Reworded commits.
len(queryset) -> queryset.count().
Rebased on master.
Contemplated if there were any three letter combination better than LES.

hmpf commented on src/argus/incident/migrations/0005_alter_event_type.py:
```diff
@@ -0,0 +1,35 @@
+            event.type="LES"
+            event.save()
+
+def remove_stateless_event_type(apps, schema_editor):
+    # undo changes made by add_stateless_event_type
+    Event = apps.get_model('argus_incident', 'Event')
+    for event in Event.objects.all():
+        if event.type == "LES":
+            event.type = "STA"
+            event.save()
```
Chef's kiss!

hmpf commented:
> Contemplated if there were any three letter combination better than LES.

I don't think there is, and I think it is three letters because that's the length of "END". The sole purpose of saving a string here instead of a key to a data table is for human convenience, and it isn't that much more expensive to have a much longer string, and not even demand the same length for each.
